elastalert/ruletypes.py

```diff
@@ -555,33 +556,53 @@ def get_all_terms(self, args):
                         level['values']['aggs'] = {'values': {'terms': copy.deepcopy(field_name)}}
                         level = level['values']['aggs']
             else:
+                self.seen_values.setdefault(field, [])
                 # For non-composite keys, only a single agg is needed
                 field_name['field'] = add_raw_postfix(field)
-            res = self.es.search(body=query, index=index, ignore_unavailable=True, timeout='50s')
-            if 'aggregations' in res:
-                buckets = res['aggregations']['filtered']['values']['buckets']
-                if type(field) == list:
-                    # For composite keys, make the lookup based on all fields
-                    # Make it a tuple since it can be hashed and used in dictionary lookups
-                    self.seen_values[tuple(field)] = []
-                    for bucket in buckets:
-                        # We need to walk down the hierarchy and obtain the value at each level
-                        self.seen_values[tuple(field)] += self.flatten_aggregation_hierarchy(bucket)
-                    # If we don't have any results, it could either be because of the absence of any baseline data
-                    # OR it may be because the composite key contained a non-primitive type.  Either way, give the
-                    # end-users a heads up to help them debug what might be going on.
-                    if not self.seen_values[tuple(field)]:
+
+            # Query the entire time range in small chunks
+            while tmp_start < end:
+                if self.rules.get('use_strftime_index'):
+                    index = format_index(self.rules['index'], tmp_start, tmp_end)
+                else:
+                    index = self.rules['index']
+                res = self.es.search(body=query, index=index, ignore_unavailable=True, timeout='50s')
+                if 'aggregations' in res:
+                    buckets = res['aggregations']['filtered']['values']['buckets']
+                    if type(field) == list:
```

Comment:
Note that there are other collections than type(..) == list, and that this will not support subclassing whatsoever. Consider isinstance(..) or issubclass(..), and checking for __iter__ and other dunder methods you intend to use, if you care about that.